File: frontend/runHubFront/src/app/models/sponsor.ts
export interface ISponsorDto {
    sponsorId?: number;
    name: string;
    logo?: string;
    description?: string;
    webPageUrl?: string;
    amount?: number;
    supportType?: string; 
  }


  export class SponsorDto implements ISponsorDto {
    sponsorId?: number;
    name: string;
    logo?: string;
    description?: string;
    webPageUrl?: string;
    amount?: number;
    supportType?: string; 

    // representatives?: Profile[];

    constructor(init: ISponsorDto) {
        this.sponsorId = init.sponsorId;
        this.name = init.name;
        this.logo = init.logo;
        this.description = init.description;
        this.webPageUrl = init.webPageUrl;
        this.amount = init.amount;
        this.supportType = init.supportType;
        // this.representatives = init.representatives ?? [];
    }
}

File: backend/RunHub.Domain/Entities/Sponsor.cs
namespace RunHub.Domain.Entity
{
    public class Sponsor
    {
        public int SponsorId { get; set; }
        public string Name { get; set; }
        public string Logo { get; set; }
        public string Description { get; set; }
        public string WebPageUrl { get; set; }
        public decimal Amount { get; set; }
        public string SupportType { get; set; }
        public int RaceId { get; set; }
        public Race Race { get; set; }
    }
}
